### DISCUSSION TITLE
How to remap keys in VI_MODE

### DISCUSSION THREAD

**Discussion by NEX-S** [xonsh/xonsh#5242] [Q&A]
hello, i want to remap L to $ in vi-normal-mode.

can anyone help me with that?

**Comment by anki-code** [comment]
hi! vi-mode is from prompt-toolkit so try to start from https://python-prompt-toolkit.readthedocs.io/en/master/pages/asking_for_input.html#vi-input-mode

Then take a look into:

https://github.com/xonsh/xonsh/blob/23c50c215e376fc5b312fa4479bf021ac7c62e80/xonsh/ptk_shell/key_bindings.py#L212

Finally ask advice in https://app.gitter.im/#/room/#xonsh_xonsh:gitter.im

**Reply by NEX-S** [reply]
thx!!!
